nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/batch/handler/impl/DefaultBatchHandlerImpl.java

```diff
@@ -0,0 +1,114 @@
+package org.nuxeo.ecm.automation.server.jaxrs.batch.handler.impl;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.Batch;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.BatchFileEntry;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.handler.AbstractBatchHandler;
+import org.nuxeo.ecm.automation.server.jaxrs.batch.handler.BatchFileInfo;
+import org.nuxeo.ecm.core.api.NuxeoException;
+import org.nuxeo.ecm.core.transientstore.api.TransientStore;
+import org.nuxeo.ecm.core.transientstore.api.TransientStoreService;
+import org.nuxeo.runtime.api.Framework;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class DefaultBatchHandlerImpl extends AbstractBatchHandler {
+
+    protected static final Log log = LogFactory.getLog(DefaultBatchHandlerImpl.class);
+
+    protected static final String BATCH_HANDLER_NAME = "default";
+
+    private static class Config {
+        public static final String TRANSIENT_STORE_NAME = "transientStore";
+    }
+
+    private TransientStore transientStore;
+    private TransientStoreService transientStoreService;
```

Comment:
Don't store framework services in fields (except for very transient objects if there is a large speed need). Don't store `transientStore` either as it's derived from it.